src/App.js

```diff
@@ -10,21 +11,72 @@ import {
   Watch,
   WatchLater,
 } from './pages';
+import { RedirectAuth, RequiresAuth } from './router';
 
 function App() {
   return (
     <>
-      <Routes>
-        <Route path='*' element={<Error />} />
-        <Route path='/' element={<Home />} />
-        <Route path='/history' element={<History />} />
-        <Route path='/liked' element={<Liked />} />
-        <Route path='/playlists' element={<Playlists />} />
-        <Route path='/watch/:id' element={<Watch />} />
-        <Route path='/watchLater' element={<WatchLater />} />
-        <Route path='/login' element={<Login />} />
-        <Route path='/signup' element={<Signup />} />
-      </Routes>
+      <Navbar />
+      <main className='container'>
+        <Sidebar />
+        <section className='pos-relative'>
+          <Routes>
+            <Route path='*' element={<Error />} />
+            <Route path='/' element={<Home />} />
+            <Route path='/watch/:id' element={<Watch />} />
+
+            <Route
+              path='/login'
+              element={
+                <RedirectAuth>
+                  <Login />
+                </RedirectAuth>
+              }
+            />
+            <Route
+              path='/signup'
+              element={
+                <RedirectAuth>
+                  <Signup />
+                </RedirectAuth>
+              }
+            />
```

Comment:
Consider having a general route handler for private routes instead, of repeating for each private route
(Both Redirect and Requires Auth)
 it will make the code easier to read and will be more understandable

Reply:
Thanks for the suggestion. 
Did the suggested changes.